src/main/scala/mesosphere/marathon/api/v2/json/Formats.scala

```diff
@@ -273,26 +255,27 @@ trait EventFormats {
   implicit lazy val GroupChangeSuccessWrites: Writes[GroupChangeSuccess] = Json.writes[GroupChangeSuccess]
   implicit lazy val GroupChangeFailedWrites: Writes[GroupChangeFailed] = Json.writes[GroupChangeFailed]
 
-  implicit lazy val DeploymentSuccessWrites: Writes[DeploymentSuccess] = Json.writes[DeploymentSuccess]
-  implicit lazy val DeploymentFailedWrites: Writes[DeploymentFailed] = Json.writes[DeploymentFailed]
-  implicit lazy val DeploymentStatusWrites: Writes[DeploymentStatus] = Json.writes[DeploymentStatus]
-  implicit lazy val DeploymentStepSuccessWrites: Writes[DeploymentStepSuccess] = Json.writes[DeploymentStepSuccess]
-  implicit lazy val DeploymentStepFailureWrites: Writes[DeploymentStepFailure] = Json.writes[DeploymentStepFailure]
-
-  implicit lazy val LightDeploymentSuccessWrites: Writes[DeploymentSuccess] = Writes { event =>
+  implicit lazy val DeploymentPlanWrites: Writes[DeploymentPlan] = Writes { plan =>
+    Json.obj(
+      "id" -> plan.id,
+      "steps" -> plan.steps,
+      "version" -> plan.version
+    )
+  }
+  implicit lazy val DeploymentSuccessWrites: Writes[DeploymentSuccess] = Writes { event =>
     Json.obj(
       "id" -> event.id,
-      "plan" -> LightDeploymentPlanWrites.writes(event.plan),
+      "plan" -> DeploymentPlanWrites.writes(event.plan),
```

Comment:
Now that there's just one implicit deployment plan writes mechanism, can we just remove DeploymentPlanWrites.writes and use the implicit?

Reply:
Good catch Tim!

I followed your suggestion and when I was cleaning up, i found that we no longer need the custom writers and thus replaced all the custom writers with the default `Json.writes[T]`. I made sure that all the "keys" in json are identical to the field names in the case classes. But lmk if this sort of implicit assumption (so that nothing breaks if someone decides to change the case class field names) is not ideal and I can revert. I see that we are doing the same thing for other case classes (like `GroupChangeFailed` for example) so I did the same for all other subclasses of `UpgradeStep`.

Also, sorry to make more commits after approval(s).